federationapi/internal/perform.go

```diff
@@ -147,27 +147,32 @@ func (r *FederationInternalAPI) performJoinUsingServer(
 	}
 
 	user, err := spec.NewUserID(userID, true)
-	if err != nil {
+	if err != nil || user == nil {
 		return err
 	}
```

Comment:
I believe we can leave out the nil check, as `NewUserID` always returns a `UserID` if the error is nil? Also in other places.

Reply:
Yup, good catch. I realized the same thing last night, but didn't fix all the locations.